Adaptors/Memory/src/SessionTable.cs

```diff
@@ -59,17 +59,31 @@ public Task Init(CancellationToken cancellationToken)
     => Task.CompletedTask;
 
   /// <inheritdoc />
-  public Task SetSessionDataAsync(string            rootSessionId,
-                                  Common.Storage.TaskOptions       defaultOptions,
-                                  CancellationToken cancellationToken = default)
+  public Task SetSessionDataAsync(string                     rootSessionId,
+                                  IEnumerable<string>        partitionIds,
+                                  Common.Storage.TaskOptions defaultOptions,
+                                  CancellationToken          cancellationToken = default)
   {
     storage_.TryAdd(rootSessionId,
                     new SessionData(rootSessionId,
                                     SessionStatus.Running,
+                                    partitionIds.ToList(),
                                     defaultOptions));
     return Task.CompletedTask;
   }
 
+  /// <inheritdoc />
+  public Task<SessionData> GetSession(string            sessionId,
+                                      CancellationToken cancellationToken = default)
+  {
+    if (!storage_.ContainsKey(sessionId))
+    {
+      throw new SessionNotFoundException($"Key '{sessionId}' not found");
+    }
+
+    return Task.FromResult(storage_[sessionId]);
```

Comment:
What is happenning if sessionId is not found?

Reply:
An exception is throwed a few lines earlier